### DISCUSSION TITLE
Accessibility and localization

### DISCUSSION THREAD

**Discussion by atmoio** [standardnotes/forum#2539] [Suggestions]
Via email: 

> A lot of the problems stem from the fact that HTML
> standards appear to have been thrown out the window. I did see you
> guys are using angular, but Angular can be made to output semantic
> HTML as well. A few examples:
> - Edit fields have no associated label. Example:
> <label for="username">Username</label> <input id="username" type="text"> etc.
> 
> - A lot of the elements on the page are clickable, but not actually
> semantically operable. I.E. clickable spans, clickable divs, clickable
> anything, but no actual buttons, links, checkboxes etc.
> - No semantic structural elements (headings, nav/main/footer tags etc.

**Comment by zersiax** [comment]
Watching :) If stuff needs to be tested, give me a shout.

**Comment by ghost** [comment]
I'm extremely interested in helping. Do you want help with a first step in making SN more semantic?

What I propose and would do is the following:

1. Make all fields have associated labels

If you are happy with my work, I could then look at transforming things like clickable headers and divs into buttons that do not look any different to the user but does make semantic sense. I would do each part extremely individually (for example, just add `<title>` to the note title field instead of` <h1>`.  Therefor any bug would be easily caught in terms of CSS, default browser behaviour, testing suite(s) and UX/UI testing.

**Comment by arielsvg** [comment]
When we start work on v4 (which we hope will be soon), we'll have to update a lot of UI components and this will be our chance to improve accessibility. I think we'll be ready to receive feedback once we ship these parts of the app :)

**Comment by ghost** [comment]
Sounds good. Is there any issue that I can help work on? I'm eager to contribute more to SN in terms of Code Review, Refactoring, Feature Additions, Pull Requests and such.

**Comment by JaspalSuri** [comment]
While you might not be able to contribute at the moment, you can familiarize yourself with or keep track of [version 004 for SNJS](https://github.com/standardnotes/snjs) (which is under active development) ahead of the release of the 3.5 update. From there it might be easier to contribute to after everything has been converted to Typescript, tested, etc.

**Comment by myreli** [comment]
Relates to the low contrast of the default `--sn-stylekit-warning-color` previously reported https://github.com/standardnotes/forum/issues/594
